### PR TITLE
Crud functionalities: PR on To-Do List App

### PR DESCRIPTION
**Pull request by AbayomiOlaoye**
- Implemented a function for adding a new task to the array.
- Implemented a function for deleting a task from the array.
- Implemented a function for editing task descriptions.
- saved all task in local storage

### CODE REVIEW

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
# Hello @AbayomiOlaoye  😄 

You're doing a great job on your project. We're proud of you. Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of GitHub Flow
  ✅  No linter errors
  ✅  Descriptive Pull Request

## Required Changes ♻️

- [ ] Index value should start from 1 instead of 0. Please fix this 👍🏻 
Screenshot 2022-10-07 at 2 43 49 PM


_Check the comments under the review._




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/to-do-list.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    this.tasks.push(task);
+    addToHTML(task, this);
+    this.updateLocalStorage();
+  };
+
+  removeTask = (toRemove) => {
+    this.tasks.splice(toRemove.index, 1);
+    this.#fixIndices(toRemove.index);
+    this.updateLocalStorage();
+  };
```
Comment (AmaduKamara) [inline src/modules/to-do-list.js:57]: [ Optional Change ] This change won't stop your project from being approved but will help improve its features.

- [ ] The delete todo feature is working but for the latest todos only. It does not delete the last to-do list. Please see what you can do about that 👍

**Review by AmaduKamara — APPROVED** [review]
# Status: APPROVED 🟢🟢🟢 

Hi @AbayomiOlaoye,

Good job on making the changes asked by the previous reviewer 👍 
Your project is complete!  

## Highlights:
✅ You used Gitflow correctly
✅ Linter checks passed
✅ Good descriptive PR 👍 
✅ Good commit messages 👍 

There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/modules/to-do-list.js (hunk 1 above)
Comment (AbayomiOlaoye) [reply src/modules/to-do-list.js:57]: Ok, thanks. I will check it.

**Pull request merged**